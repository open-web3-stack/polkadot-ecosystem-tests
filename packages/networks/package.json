{
	"name": "@e2e-test/networks",
	"type": "module",
	"dependencies": {
		"@acala-network/chopsticks-testing": "^0.13.1",
		"@e2e-test/shared": "workspace:*"
	},
	"main": "./dist/cjs/index.js",
	"module": "./dist/esm/index.js",
	"files": [
		"*",
		"!tsconfig.json",
		"!**/*.test.ts"
	],
	"exports": {
		".": {
			"require": "./dist/cjs/index.js",
			"import": "./dist/esm/index.js",
			"default": "./dist/esm/index.js"
		},
		"./chains": {
			"require": "./dist/cjs/chains/index.js",
			"import": "./dist/esm/chains/index.js",
			"default": "./dist/esm/chains/index.js"
		},
		"./*": {
			"require": "./dist/cjs/*.js",
			"import": "./dist/esm/*.js",
			"default": "./dist/esm/*.js"
		}
	}
}
