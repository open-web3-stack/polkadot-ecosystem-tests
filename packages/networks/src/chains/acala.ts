import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  acala: {
    relayToken: 'DOT',
    relayLiquidToken: 'LDOT',
    stableToken: 'AUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    dot: { Token: 'DOT' },
    ldot: { Token: 'LDOT' },
    dai: { Erc20: '0x54a37a01cd75b616d63e0ab665bffdb0143c52ae' },
    wbtc: { ForeignAsset: 5 },
    ausd: { Token: 'AUSD' },
    aca: { Token: 'ACA' },
    lcdot: { LiquidCrowdloan: 13 },
  },
  karura: {
    relayToken: 'KSM',
    relayLiquidToken: 'LKSM',
    stableToken: 'KUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    ksm: { Token: 'KSM' },
    lksm: { Token: 'LKSM' },
    usdt: { ForeignAsset: 7 },
    rmrk: { ForeignAsset: 0 },
    dai: { Erc20: '0x4bb6afb5fa2b07a5d1c499e1c3ddb5a15e709a71' },
    ausd: { Token: 'KUSD' },
    kar: { Token: 'KAR' },
  },
}

const getInitStorages = (config: typeof custom.acala | typeof custom.karura) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 4, data: { free: 10 * 1e12 } }]],
  },
  Tokens: {
    accounts: [
      [[defaultAccount.alice.address, { Token: config.relayToken }], { free: 10 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.relayLiquidToken }], { free: 100 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.stableToken }], { free: 1000 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccount.alice.address,
  },
  EvmAccounts: {
    accounts: [[['0x82a258cb20e2adb4788153cd5eb5839615ece9a0'], defaultAccount.alice.address]],
    evmAddresses: [[[defaultAccount.alice.address], '0x82a258cb20e2adb4788153cd5eb5839615ece9a0']],
  },
  Homa: {
    // avoid impact test outcome
    $removePrefix: ['redeemRequests', 'unbondings', 'toBondPool'],
    // so that bump era won't trigger unbond
    relayChainCurrentEra: 100,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const acala = defineChain({
  name: 'acala',
  endpoint: 'wss://acala-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.acala,
  initStorages: getInitStorages(custom.acala),
})

export const karura = defineChain({
  name: 'karura',
  endpoint: 'wss://karura-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.karura,
  initStorages: getInitStorages(custom.karura),
})
