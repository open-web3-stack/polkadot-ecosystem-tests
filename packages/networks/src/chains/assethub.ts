import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
  assetHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubPolkadot | typeof custom.assetHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubPolkadot = defineChain({
  name: 'assetHubPolkadot',
  endpoint: 'wss://rpc-asset-hub-polkadot.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubPolkadot,
  initStorages: getInitStorages(custom.assetHubPolkadot),
})

export const assetHubKusama = defineChain({
  name: 'assetHubKusama',
  endpoint: 'wss://rpc-asset-hub-kusama.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubKusama,
  initStorages: getInitStorages(custom.assetHubKusama),
})
