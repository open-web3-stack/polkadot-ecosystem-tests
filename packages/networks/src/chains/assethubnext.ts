import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNextWestend: {
    wnd: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNextWestend) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNextWestend = defineChain({
  name: 'assetHubNextWestend',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNextWestend,
  initStorages: getInitStorages(custom.assetHubNextWestend),
})
