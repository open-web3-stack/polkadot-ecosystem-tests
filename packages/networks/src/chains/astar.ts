import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  astar: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551617',

    paraAccount: '13YMK2eZzuFY1WZGagpYtTgbWBWGdoUD2CtrPj1mQPjY8Ldc',
    dot: 340282366920938463463374607431768211455n,
    astr: { Concrete: { parents: 0, interior: 'Here' } },
    aca: 18446744073709551616n,
    usdt: 4294969280n,
  },
  shiden: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551616',

    paraAccount: 'F7fq1jNy74AqkJ1DP4KqSrWtnTGtXfNVoDwFhTvvPxUvJaq',
    ksm: 340282366920938463463374607431768211455n,
    sdn: { Concrete: { parents: 0, interior: 'Here' } },
    kar: 18446744073709551618n,
    usdt: 4294969280n,
  },
}

const getInitStorages = (config: typeof custom.astar | typeof custom.shiden) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: '100000000000000000000' } }]],
  },
  Assets: {
    account: [
      [[config.relayToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
      [[config.aUSDToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccounts.alice.address,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const astar = defineChain({
  name: 'astar',
  paraId: 2006,
  endpoint: 'wss://astar-rpc.dwellir.com',
  custom: custom.astar,
  initStorages: getInitStorages(custom.astar),
})

export const shiden = defineChain({
  name: 'shiden',
  paraId: 2007,
  endpoint: 'wss://shiden-rpc.dwellir.com',
  custom: custom.shiden,
  initStorages: getInitStorages(custom.shiden),
})
