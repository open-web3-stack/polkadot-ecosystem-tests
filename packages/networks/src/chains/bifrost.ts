import { defineChain } from '../defineChain.js'
import { defaultAccounts, defaultAccountsSr25519, testAccounts } from '../testAccounts.js'

const custom = {
  bifrostPolkadot: {
    relayToken: 'DOT',
    bnc: { Native: 'BNC' },
  },
  bifrostKusama: {
    relayToken: 'KSM',
    bnc: { Token: 'BNC' },
  },
}

const getInitStorages = (_config: typeof custom.bifrostPolkadot | typeof custom.bifrostKusama) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[testAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
    ],
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers'],
  },
})

export const bifrostPolkadot = defineChain({
  name: 'bifrostPolkadot',
  endpoint: 'wss://bifrost-polkadot.ibp.network',
  paraId: 2030,
  custom: custom.bifrostPolkadot,
  initStorages: getInitStorages(custom.bifrostPolkadot),
})

export const bifrostKusama = defineChain({
  name: 'bifrostKusama',
  endpoint: 'wss://us.bifrost-rpc.liebi.com/ws',
  paraId: 2001,
  custom: custom.bifrostKusama,
  initStorages: getInitStorages(custom.bifrostKusama),
})
