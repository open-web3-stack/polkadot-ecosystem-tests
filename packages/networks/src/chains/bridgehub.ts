import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  bridgeHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
  },
  bridgeHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
  },
}

const getInitStorages = (_config: typeof custom.bridgeHubPolkadot | typeof custom.bridgeHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
})

export const bridgeHubPolkadot = defineChain({
  name: 'bridgeHubPolkadot',
  endpoint: 'wss://polkadot-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubPolkadot,
  initStorages: getInitStorages(custom.bridgeHubPolkadot),
})

export const bridgeHubKusama = defineChain({
  name: 'bridgeHubKusama',
  endpoint: 'wss://kusama-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubKusama,
  initStorages: getInitStorages(custom.bridgeHubKusama),
})
