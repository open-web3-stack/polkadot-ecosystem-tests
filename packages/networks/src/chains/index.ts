export * from './acala.js'
export * from './assethub.js'
export * from './astar.js'
export * from './hydraDX.js'
export * from './moonbeam.js'
export * from './polkadot.js'
