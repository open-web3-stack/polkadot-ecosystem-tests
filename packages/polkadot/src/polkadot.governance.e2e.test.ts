import { polkadot } from '@e2e-test/networks/chains'

import { governanceE2ETests } from '@e2e-test/shared'

governanceE2ETests(polkadot, { testSuiteName: 'Polkadot Governance', addressEncoding: 0 })
