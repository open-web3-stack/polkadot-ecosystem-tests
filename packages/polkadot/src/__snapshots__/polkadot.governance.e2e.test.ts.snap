// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Polkadot Governance > preimage submission, query and removal works > note preimage events 1`] = `
[
  {
    "data": {
      "hash_": "(hash)",
    },
    "method": "Noted",
    "section": "preimage",
  },
]
`;

exports[`Polkadot Governance > preimage submission, query and removal works > unnote preimage events 1`] = `
[
  {
    "data": {
      "hash_": "(hash)",
    },
    "method": "Cleared",
    "section": "preimage",
  },
]
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > cancelling referendum with signed origin 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 84000)",
          "refTime": "(rounded 560000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > charlie's class locks after their vote's rescission 1`] = `
[
  [
    30,
    50000000000,
  ],
]
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > charlie's votes after casting his 1`] = `
{
  "standard": {
    "balance": 50000000000,
    "vote": "0x83",
  },
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > charlie's votes after referendum's cancellation 1`] = `
{
  "standard": {
    "balance": 50000000000,
    "vote": "0x83",
  },
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > charlie's votes after rescission 1`] = `
{
  "delegations": {
    "capital": 0,
    "votes": 0,
  },
  "prior": [
    0,
    0,
  ],
  "votes": [],
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > dave's class locks after their vote's rescission 1`] = `
[
  [
    30,
    60000000000,
  ],
]
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > dave's votes after casting his 1`] = `
{
  "split": {
    "aye": 50000000000,
    "nay": 10000000000,
  },
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > dave's votes after referendum's cancellation 1`] = `
{
  "split": {
    "aye": 50000000000,
    "nay": 10000000000,
  },
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > dave's votes after rescission 1`] = `
{
  "delegations": {
    "capital": 0,
    "votes": 0,
  },
  "prior": [
    0,
    0,
  ],
  "votes": [],
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > eve's class locks after their vote's rescission 1`] = `
[
  [
    30,
    80000000000,
  ],
]
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > eve's votes after casting hers 1`] = `
{
  "splitAbstain": {
    "abstain": 20000000000,
    "aye": 50000000000,
    "nay": 10000000000,
  },
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > eve's votes after referendum's cancellation 1`] = `
{
  "splitAbstain": {
    "abstain": 20000000000,
    "aye": 50000000000,
    "nay": 10000000000,
  },
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > eve's votes after rescission 1`] = `
{
  "delegations": {
    "capital": 0,
    "votes": 0,
  },
  "prior": [
    0,
    0,
  ],
  "votes": [],
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > events for bob's decision deposit 1`] = `
[
  {
    "data": {
      "amount": 10000000000,
      "who": "14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3",
    },
    "method": "DecisionDepositPlaced",
    "section": "referenda",
  },
]
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > events for charlie's vote 1`] = `
[
  {
    "data": {
      "vote": {
        "Standard": {
          "balance": 50000000000,
          "vote": {
            "conviction": "Locked3x",
            "vote": "Aye",
          },
        },
      },
      "who": "14Gjs1TD93gnwEBfDMHoCgsuf1s2TVKUP6Z1qKmAZnZ8cW5q",
    },
    "method": "Voted",
    "section": "convictionVoting",
  },
]
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > events for dave's vote 1`] = `
[
  {
    "data": {
      "vote": {
        "Split": {
          "aye": 50000000000,
          "nay": 10000000000,
        },
      },
      "who": "126TwBzBM4jUEK2gTphmW4oLoBWWnYvPp8hygmduTr4uds57",
    },
    "method": "Voted",
    "section": "convictionVoting",
  },
]
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > events for eve's vote 1`] = `
[
  {
    "data": {
      "vote": {
        "SplitAbstain": {
          "abstain": 20000000000,
          "aye": 50000000000,
          "nay": 10000000000,
        },
      },
      "who": "16D2eVuK5SWfwvtFD3gVdBC2nc2BafK31BY6PrbZHBAGew7L",
    },
    "method": "Voted",
    "section": "convictionVoting",
  },
]
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > referendum info after charlie's vote 1`] = `
{
  "ongoing": {
    "deciding": {
      "confirming": null,
    },
    "decisionDeposit": {
      "amount": 10000000000,
      "who": "14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3",
    },
    "enactment": {
      "after": 1,
    },
    "inQueue": false,
    "origin": {
      "origins": "SmallTipper",
    },
    "proposal": {
      "inline": "0x13030700e40b5402008eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48",
    },
    "submissionDeposit": {
      "amount": 10000000000,
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "tally": {
      "ayes": 150000000000,
      "nays": 0,
      "support": 50000000000,
    },
    "track": 30,
  },
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > referendum info after dave's vote 1`] = `
{
  "ongoing": {
    "deciding": {
      "confirming": null,
    },
    "decisionDeposit": {
      "amount": 10000000000,
      "who": "14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3",
    },
    "enactment": {
      "after": 1,
    },
    "inQueue": false,
    "origin": {
      "origins": "SmallTipper",
    },
    "proposal": {
      "inline": "0x13030700e40b5402008eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48",
    },
    "submissionDeposit": {
      "amount": 10000000000,
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "tally": {
      "ayes": 155000000000,
      "nays": 1000000000,
      "support": 100000000000,
    },
    "track": 30,
  },
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > referendum info after eve's vote 1`] = `
{
  "ongoing": {
    "deciding": {
      "confirming": null,
    },
    "decisionDeposit": {
      "amount": 10000000000,
      "who": "14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3",
    },
    "enactment": {
      "after": 1,
    },
    "inQueue": false,
    "origin": {
      "origins": "SmallTipper",
    },
    "proposal": {
      "inline": "0x13030700e40b5402008eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48",
    },
    "submissionDeposit": {
      "amount": 10000000000,
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "tally": {
      "ayes": 160000000000,
      "nays": 2000000000,
      "support": 170000000000,
    },
    "track": 30,
  },
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > referendum info before decision deposit 1`] = `
{
  "ongoing": {
    "deciding": null,
    "decisionDeposit": null,
    "enactment": {
      "after": 1,
    },
    "inQueue": false,
    "origin": {
      "origins": "SmallTipper",
    },
    "proposal": {
      "inline": "0x13030700e40b5402008eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48",
    },
    "submissionDeposit": {
      "amount": 10000000000,
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "tally": {
      "ayes": 0,
      "nays": 0,
      "support": 0,
    },
    "track": 30,
  },
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > referendum info post decision deposit 1`] = `
{
  "ongoing": {
    "deciding": null,
    "decisionDeposit": {
      "amount": 10000000000,
      "who": "14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3",
    },
    "enactment": {
      "after": 1,
    },
    "inQueue": false,
    "origin": {
      "origins": "SmallTipper",
    },
    "proposal": {
      "inline": "0x13030700e40b5402008eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48",
    },
    "submissionDeposit": {
      "amount": 10000000000,
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "tally": {
      "ayes": 0,
      "nays": 0,
      "support": 0,
    },
    "track": 30,
  },
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > referendum submission events 1`] = `
[
  {
    "data": {
      "proposal": {
        "Inline": "(hash)",
      },
      "track": 30,
    },
    "method": "Submitted",
    "section": "referenda",
  },
]
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > referendum upon start of decision period 1`] = `
{
  "deciding": {
    "confirming": null,
  },
  "decisionDeposit": {
    "amount": 10000000000,
    "who": "14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3",
  },
  "enactment": {
    "after": 1,
  },
  "inQueue": false,
  "origin": {
    "origins": "SmallTipper",
  },
  "proposal": {
    "inline": "0x13030700e40b5402008eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48",
  },
  "submissionDeposit": {
    "amount": 10000000000,
    "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
  },
  "tally": {
    "ayes": 0,
    "nays": 0,
    "support": 0,
  },
  "track": 30,
}
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > refund of decision deposit 1`] = `
[
  {
    "data": {
      "amount": 10000000000,
      "who": "14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3",
    },
    "method": "DecisionDepositRefunded",
    "section": "referenda",
  },
]
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > refund of submission deposit 1`] = `
[
  {
    "data": {
      "amount": 10000000000,
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "method": "SubmissionDepositRefunded",
    "section": "referenda",
  },
]
`;

exports[`Polkadot Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > removal of votes in cancelled referendum 1`] = `
[
  {
    "data": {
      "amount": "(rounded 220000000)",
    },
    "method": "Withdraw",
    "section": "balances",
  },
  {
    "data": {},
    "method": "ItemCompleted",
    "section": "utility",
  },
  {
    "data": {},
    "method": "ItemCompleted",
    "section": "utility",
  },
  {
    "data": {},
    "method": "ItemCompleted",
    "section": "utility",
  },
  {
    "data": {},
    "method": "BatchCompleted",
    "section": "utility",
  },
  {
    "data": {
      "amount": "(rounded 180000000)",
    },
    "method": "Deposit",
    "section": "balances",
  },
  {
    "data": {
      "amount": "(rounded 44000000)",
    },
    "method": "Deposit",
    "section": "balances",
  },
  {
    "data": {
      "actualFee": "(rounded 220000000)",
      "tip": 0,
    },
    "method": "TransactionFeePaid",
    "section": "transactionPayment",
  },
  {
    "data": {
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 150000)",
          "refTime": "(rounded 1400000000)",
        },
      },
    },
    "method": "ExtrinsicSuccess",
    "section": "system",
  },
]
`;

exports[`Polkadot Governance > referendum lifecycle test 2 - submission, decision deposit, and killing should work > killing referendum with signed origin 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 84000)",
          "refTime": "(rounded 650000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;
